### DISCUSSION TITLE
Is it possible to add custom widget that extends ReactWidget in plugin?

### DISCUSSION THREAD

**Discussion by eli88p** [eclipse-theia/theia#9204] [Q&A]
### Description
I tried to build UI in plugin it's possible to render tsx files in plugin?

Thanks in advance.

**Comment by akosyakov** [comment]
@eli88p No, plugins are isolated from the main window code. You can still use React with VS Code webviews.

What is your use case? Do you plan to build a custom product or do you want to provide a single VS Code extension and reuse between VS Code and Theia? Webviews are limited since they don't have access to the core application and treated as a remote content. If you want to tailor UI then you should go with Theia extension or a hybrid, i.e. using Theia and VS Code extensions together and communicate between via commands.

**Comment by eli88p** [comment]
@akosyakov Thanks
